### PR TITLE
feat: add unit tests for schema, functions and introspection

### PR DESCRIPTION
**Pull request by vi3k6i5**
feat: add unit tests for schema, functions and introspection updated nox file for coverage and settings path for running unit tests.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -101,3 +124,40 @@ def docs(session):
...
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def docfx(session):
+    """Build the docfx yaml files for this library."""
+
+    session.install("-e", ".[tracing]")
+    # sphinx-docfx-yaml supports up to sphinx version 1.5.5.
+    # https://github.com/docascode/sphinx-docfx-yaml/issues/97
+    session.install(
+        "sphinx==1.5.5", "alabaster", "recommonmark", "sphinx-docfx-yaml"
+    )
```
Comment (busunkim96) [inline noxfile.py:138]: Thanks for adding this! Small tweak to match the current template: https://github.com/googleapis/synthtool/blob/063de45298fbdd88916018ba566c7ecd254b39ae/synthtool/gcp/templates/python_library/noxfile.py.j2#L260

```suggestion
    session.install(
        "sphinx", "alabaster", "recommonmark", "gcp-sphinx-docfx-yaml"
    )
```

Comment (vi3k6i5) [reply noxfile.py]: made the change, thanks :)

**Review by thiagotnunes — APPROVED** [review]
LGTM, but please wait for another approver since my python knowledge is limited

**Comment by vi3k6i5** [comment]
Splitting this into 3 parts:
1. https://github.com/googleapis/python-spanner-django/pull/610 [ nox file config for docs]
2. https://github.com/googleapis/python-spanner-django/pull/611 [added unit tests with coverage 75 %]
3. Pending need to fix the settings file for it.

**Comment by vi3k6i5** [comment]
Split this PR into multiple PRs, so closing it.

**Pull request closed without merging**